File: 33_配列/go/main.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

var r *rand.Rand

func init() {
	r = rand.New(rand.NewSource(time.Now().UnixNano()))
}

func generateRamdomIndex(max int) int {
	return r.Int() % max
}

func main() {

	answers := []string{"Yes", "No", "Maybe", "Ask again later"}
	readUserInput("What's your question? ")
	answer := answers[generateRamdomIndex(len(answers))]
	fmt.Printf("%s.\n", answer)
}

File: 33_配列/rust/src/main.rs
extern crate rand;

use rand::{thread_rng, Rng};
use std::io::prelude::*;

fn main() -> std::io::Result<()> {
    print!("What's your question? ");
    std::io::stdout().flush()?;

    let mut buf = String::new();
    std::io::stdin().read_line(&mut buf)?;

    let answers = vec!["Yes", "No", "Maybe", "Ask again later"];

    let num = thread_rng().gen_range(0, answers.len());
    // println!("{}", answers[num]);
    let answer = answers[num];
    println!("{}", answer);

    Ok(())
}
